packages/engine/src/datastore/table/context/mod.rs

```diff
@@ -161,10 +157,24 @@ impl Context {
     }
 }
 
+impl Deref for Context {
```

Comment:
I'm not sure if we want to equate context with the context batch. For example, the messages column in the message pool of the context is quite different from the messages column of the context batch. The first has the actual contents of the previous state's messages, whereas the second has tuples of indices referring to where an agent's messages are in memory.

This might mean that the context batch should be renamed or just that we should rename other batches to groups as we discussed earlier.

@Alfred-Mountfield thoughts?

Reply:
I think as you say, we should rename the `ContextBatch`, because it's pretty unclear what portions of the conceptual "Context" it actually includes.

We should also rename the batches and figure out groups as we've said. We've got a task to track that so I think in the meantime we should decide if we want to rename the `ContextBatch` now, and what to rename it to. Lets discuss off GH and then post an update here.

(I also think we need to clarify the message columns)

Reply:
We're including it in the scope of the [task](https://app.asana.com/0/1199548034582004/1201800853370429/f) for clarifying groups and batches

Reply:
I'd go with just `global` to avoid "batch.batch", but either is fine.

Reply:
I mean that we'd have `Context.global_batch() -> Batch` so you would end up with `batch.batch`

Reply:
I removed the `Deref` and `DerefMut` implementation and replaced it by `global_batch` and `global_batch_mut`.